src/Forms/FormField.php

```diff
@@ -1221,18 +1221,29 @@ public function Type()
         return strtolower(preg_replace('/Field$/', '', $type->getShortName() ?? '') ?? '');
     }
 
+    /**
+     * Utility method to call an extension hook which allows the result of validate() calls to be adjusted
+     *
+     * @param bool $result
+     * @param Validator $validator
+     * @return bool
+     */
+    protected function extendValidationResult($result, $validator)
+    {
+        $this->extend('updateValidationResult', $result, $validator);
+        return $result;
+    }
```

Comment:
This is broken out into a separate method to save a bunch of duplication across FormField subclasses like:

```php
$result = true;
$this->extend('updateValidationResult', $result, $validator);
return $result;
```

For some fields (like `ConfirmedPasswordField`) this would end up being copy-pasted many times in one method, so it felt better to break it out like this.

The only other alternative I could spot would be to refactor all the `validate()` methods to take this approach:

```php
$result = true;
if (failedValidation()) {
    $result = false;
}
$this->extend('updateValidationResult', $result, $validator);
return $result;
```

That works for simple fields (I essentially did that for `TextField` for example) but for `ConfirmedPasswordField` it means a big refactor and removing a bunch of early-returns, which just felt like a lot of work for no real gain in readability.

Reply:
This is a great solution - makes it really easy for third-party formfields to use this functionality.